cheddar/cheddar-server/src/main/java/com/clicktravel/cheddar/server/rest/application/RestApplication.java

```diff
@@ -51,20 +51,30 @@ public static void main(final String... args) throws Exception {
         SLF4JBridgeHandler.removeHandlersForRootLogger();
         SLF4JBridgeHandler.install();
         final Logger logger = LoggerFactory.getLogger(RestApplication.class);
-        logger.debug(String.format("java.version:[%s] java.vendor:[%s]", System.getProperty("java.version"),
-                System.getProperty("java.vendor")));
-        final RestServer restServer = new RestServer(servicePort, statusPort, bindAddress);
-        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
-            @Override
-            public void run() {
-                restServer.stop();
-            }
-        }));
         try {
-            restServer.start();
+            logger.info("Java process starting");
+            logger.debug(String.format("java.version:[%s] java.vendor:[%s]", System.getProperty("java.version"),
+                    System.getProperty("java.vendor")));
+            final ConfigurableApplicationContext applicationContext = new ClassPathXmlApplicationContext(
+                    "applicationContext.xml");
+            logger.debug("Finished getting ApplicationContext");
+            final ApplicationLifecycleController applicationLifecycleController = applicationContext
+                    .getBean(ApplicationLifecycleController.class);
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                logger.info("Shutdown hook invoked - Commencing graceful termination of Java process");
+                applicationLifecycleController.shutdownApplication();
```

Comment:
Please note I have updated method `MessageRoutingProcessor#shutdown` to initiate shutdown and then await for shutdown completion of all message listeners. Similar shutdown logic is implemented in method `ApplicationLifecycleController#shutdownApplication`